crypto/curve25519/curve25519.c

```diff
@@ -261,27 +263,192 @@ static void x25519_s2n_bignum_public_from_private(
 #endif
 }
 
-// Stub function until ED25519 lands in s2n-bignum
 static void ed25519_public_key_from_hashed_seed_s2n_bignum(
   uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
   uint8_t az[SHA512_DIGEST_LENGTH]) {
+
```

Comment:
I attempted to factor out as much as possible. The remaining "duplication" is due to using different functions working on different internal representations. For example, this can illustrate it:
nohw path:
```
  // Reduce r modulo the order of the base-point B.
  x25519_sc_reduce(r);
  ge_p3 R;
  // Compute [r]B.
  x25519_ge_scalarmult_base(&R, r);
```
while s2n-bignum patch:
```
  uint64_t R[8] = {0};  
  // Reduce r modulo the order of the base-point B.
  bignum_mod_n25519(uint64_r, 8, uint64_r);

  // Compute [r]B.
  scalarmulbase(R, uint64_r);
```
so, in former case R is some internal representation of a group element. In the latter case, R is an uin64_t  buffer.

I think in this case, having a bit of duplication is actually more read-friendly than not having it. Hopefully, in the future we can just have a high-level s2n-bignum function that would remove this duplication.